Makefile

```diff
@@ -127,8 +149,16 @@ $(WEAVE_EXPORT): $(IMAGES_UPTODATE)
 $(DOCKER_DISTRIB):
 	curl -o $(DOCKER_DISTRIB) $(DOCKER_DISTRIB_URL)
 
-tests: tools/.git
-	tools/test
+ifeq ($(BUILD_IN_CONTAINER),true)
+tests: $(BUILD_UPTODATE) $(VENDOR_UPTODATE) tools/.git
+	$(SUDO) docker run $(RM) $(RUN_FLAGS) \
+		-v $(shell pwd):/go/src/github.com/weaveworks/weave \
+		-e GOARCH -e GOOS -e CIRCLECI -e CIRCLE_BUILD_NUM -e CIRCLE_NODE_TOTAL -e CIRCLE_NODE_INDEX -e COVERDIR\
```

Comment:
those circleci env vars are annoying. Nothing else in the makefile makes any reference to circleci.